File: Serene/Serene.Web/Modules/Membership/Account/SignUp/SignUpPanel.ts
namespace Serene.Membership {

    @Serenity.Decorators.registerClass()
    export class SignUpPanel extends Serenity.PropertyPanel<SignUpRequest, any> {

        protected getFormKey() { return SignUpForm.formKey; }

        private form: SignUpForm;

        constructor(container: JQuery) {
            super(container);

            this.form = new SignUpForm(this.idPrefix);

            this.form.ConfirmEmail.addValidationRule(this.uniqueName, e => {
                if (this.form.ConfirmEmail.value !== this.form.Email.value) {
                    return Q.text('Validation.EmailConfirm');
                }
            });

            this.form.ConfirmPassword.addValidationRule(this.uniqueName, e => {
                if (this.form.ConfirmPassword.value !== this.form.Password.value) {
                    return Q.text('Validation.PasswordConfirm');
                }
            });

            this.byId('SubmitButton').click(e => {
                e.preventDefault();

                if (!this.validateForm()) {
                    return;
                }

                Q.serviceCall({
                    url: Q.resolveUrl('~/Account/SignUp'),
                    request: {
                        DisplayName: this.form.DisplayName.value,
                        Email: this.form.Email.value,
                        Password: this.form.Password.value
                    },
                    onSuccess: response => {
                        Q.information(Q.text('Forms.Membership.SignUp.Success'), () => {
                            window.location.href = Q.resolveUrl('~/');
                        });
                    }
                });

            });
        }
    }
}

File: Serene/Serene.Script/Membership/SignUpPanel.cs

namespace Serene.Membership
{
    using jQueryApi;
    using Serenity;
    using System.Html;
    using System.Runtime.CompilerServices;

    [Imported, Panel, FormKey("Membership.SignUp")]
    public class SignUpPanel : PropertyPanel<SignUpRequest>
    {
        private SignUpForm form;

        public SignUpPanel(jQueryObject container)
            : base(container)
        {
            form = new SignUpForm(this.IdPrefix);

            form.ConfirmPassword.AddValidationRule(this.uniqueName, e =>
            {
                if (form.ConfirmPassword.Value != form.Password.Value)
                    return Q.Text("Validation.PasswordConfirm");

                return null;
            });

            form.ConfirmEmail.AddValidationRule(this.uniqueName, e =>
            {
                if (form.ConfirmEmail.Value != form.Email.Value)
                    return Q.Text("Validation.EmailConfirm");

                return null;
            });

            this.ById("SubmitButton").Click((s, e) =>
            {
                e.PreventDefault();

                if (!ValidateForm())
                    return;

                Q.ServiceCall(new ServiceCallOptions
                {
                    Url = Q.ResolveUrl("~/Account/SignUp"),
                    Request = new SignUpRequest
                    {
                        DisplayName = form.DisplayName.Value,
                        Email = form.Email.Value,
                        Password = form.Password.Value
                    },
                    OnSuccess = response =>
                    {
                        Q.Information(Q.Text("Forms.Membership.SignUp.Success"), () =>
                        {
                            Window.Location.Href = Q.ResolveUrl("~/");
                        }, new ConfirmOptions());
                    }
                });
            });
        }
    }
}